subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JavaCompilerArgumentsBuilder.java

```diff
@@ -135,9 +135,13 @@ private void addMainOptions(List<String> compilerArgs) {
         if (!includeMainOptions) {
             return;
         }
-
```

Comment:
Should `validateCompilerArgs` fail if you specify the release property and `--release`?
And I wonder if `JdkJavaCompiler` should actually check if `--release` is even supported. Or should we just rely on the compiler bailing if it is not supported?

Reply:
Nonetheless, I think a test using `AvailableJavaHomes.getJdk8` to determine that a user gets a proper error message (no matter who errors out)

Reply:
Yes, I have added an error case for combining `--release` and `JavaCompile.release`.

For the part about Gradle bailing if the target VM does not support it, to do in a follow up PR. It is planned indeed.